src/core/path.c

```diff
@@ -342,12 +342,18 @@ static int path_add_default_dependencies(Path *p) {
                 return r;
 
         if (MANAGER_IS_SYSTEM(UNIT(p)->manager)) {
-                r = unit_add_two_dependencies_by_name(UNIT(p), UNIT_AFTER, UNIT_REQUIRES, SPECIAL_SYSINIT_TARGET, true, UNIT_DEPENDENCY_DEFAULT);
+                r = unit_add_two_dependencies_by_name(
+                                UNIT(p),
+                                UNIT_AFTER,
+                                UNIT(p)->survive ? _UNIT_DEPENDENCY_INVALID : UNIT_REQUIRES,
```

Comment:
i don't get this? why should this boolean downgrade the from After=sysinit.target + Requires=sysinit.target to just After=sysinit.target in case DefaultDependencies=no is set?

soft-reboot surviving units will by definition run very early and very late at boot, so why would you order them against sysinit.target? that's what you do for units that shall only run during regular runtime, i.e. start late enough and stop early enough.

for services that shall survive that long there's already a knob: DefaultDependencies=no, so why this change here?

Reply:
Because when sysinit goes away, the unit would go away too due to the Require. We want ordering here, but not dependency.

Reply:
do we want ordering here? that's not obvious to me. soft-reboot services want to survive until the very end, and then are already running at the very beginning of the next boot. hence ordering is really weird for them.

Reply:
As below, ordering doesn't matter when a unit is already running, it's a no-op, that's already the case today. But on 'real' boot it matters a lot. Just because something is configured to survive a system transition, it doesn't imply that on a real boot it should start super early at boot, nor that on a real shutdown it should stay alive until very late. These are very orthogonal concepts - and if needed, they can be configured separately.